### PR TITLE
Add HyperOne builder

### PR DESCRIPTION
**Pull request by m110**
This PR adds a new builder for HyperOne platform: http://www.hyperone.com/ 

It includes acceptance tests and documentation. Support for building chroot disks is loosely based on the Amazon Chroot builder.

Example template:

``` json
{
    "type": "hyperone",
    "token": "YOUR_AUTH_TOKEN",
    "source_image": "ubuntu",
    "vm_type": "a1.nano",
    "disk_size": 10
}
```

### CODE REVIEW

**Comment by ad-m** [comment]
@m101, build failed:
```
7.91s$ make ci
==> Checking that code complies with gofmt requirements...
Check passed.
find: -executable: unknown primary or operator
find: -executable: unknown primary or operator
==> Checking that only certain files are executable...
Check passed.
vendor/github.com/hyperonecom/h1-client-go/api_container.go:19:2: cannot find package "github.com/antihax/optional" in any of:
	/Users/travis/gopath/src/github.com/hashicorp/packer/vendor/github.com/antihax/optional (vendor tree)
	/Users/travis/.gimme/versions/go1.11.5.darwin.amd64/src/github.com/antihax/optional (from $GOROOT)
	/Users/travis/gopath/src/github.com/antihax/optional (from $GOPATH)
make: *** [testrace] Error 1
```

**Comment by ad-m** [comment]
What remains to be done for the changes to be merged? What additional conditions must be met?

Integration with Hashicorp tools is important for HyperOne. For this reason, we intend to conduct a daily test with the Packer's "master" version to ensure that it will work now and in the future. There is scheduler in https://github.com/hyperonecom/h1-packer-e2e to rebuild every day. We also already use it in internal processes of prepare "Recommended images". There is required tooling: https://github.com/hyperonecom/h1-images-recommended . This proves the effectiveness and usefulness of the developed integration for Packer users.

**Comment by SwampDragons** [comment]
I just need to review the code and then we can merge. I intend to spend a lot of time this week on PRs so I should be able to get to it fairly soon.

**Comment by gregorybrzeski** [comment]
@SwampDragons did you managed to have a look at this code ? Do you have any feedback so we can adjust it accordingly ?

**Comment by SwampDragons** [comment]
I'm so sorry; I've been out most of the week with the flu. I'll probably be back Monday and I'll review then, I promise 😬

File: builder/hyperone/step_create_image.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		return multistep.ActionHalt
+	}
+
+	state.Put("image_id", image.Id)
+	state.Put("image_name", image.Name)
+
+	return multistep.ActionContinue
+}
+
+func (s *stepCreateImage) Cleanup(state multistep.StateBag) {}
```
Comment (SwampDragons) [inline builder/hyperone/step_create_image.go]: You may want to add an error state check here and clean up the image by deleting it if the Packer build failed.

Comment (m110) [reply builder/hyperone/step_create_image.go]: Thank you for pointing that out. I've added a cleanup step in 0673bb2c57aa88b360aa6fc1b06c0904ea47cab8

**Comment by gregorybrzeski** [comment]
@SwampDragons I can confirm that we are happy to maintain this code over time and respond to issues if they arise. We are prepared for long term commitment to HashiCorp tools and starting work on Terraform provider integration as well. 

We have managed to:

- add reviewers to CODEOWNERS file
- added cleanup step as suggested
- updated code from master for all tests to pass

I hope that's all what's needed to merge.

**Comment by SwampDragons** [comment]
Indeed!  Thanks for your patience on this.

**Pull request merged**